### PR TITLE
Replace logical name by name

### PR DESCRIPTION
**Pull request by veolys-nanne**
[OO-FEATURE] allow overriding logical name in route generating by using  route name

### CODE REVIEW

File: FrontBundle/Routing/Database/Transformer/RouteDocumentToValueObjectTransformer.php [hunk 1]
```diff
@@ -4,20 +4,34 @@
...
+use OpenOrchestra\FrontBundle\Exception\NoRenderingMethodForNodeException;
 
 /**
  * Class RouteDocumentToValueObjectTransformer
  */
 class RouteDocumentToValueObjectTransformer
 {
+    protected $logicalName;
+
+    public function __construct(RouterInterface $router)
```
Comment (ngilain) [inline FrontBundle/Routing/Database/Transformer/RouteDocumentToValueObjectTransformer.php:20]: @itkg-nanne : phpdoc missing

**Comment by ngilain** [comment]
:+1: when the phpdoc updated

File: FrontBundle/Tests/Routing/Database/Transformer/RouteDocumentToValueObjectTransformerTest.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 use OpenOrchestra\BaseBundle\Tests\AbstractTest\AbstractBaseTestCase;
 use OpenOrchestra\FrontBundle\Routing\Database\Transformer\RouteDocumentToValueObjectTransformer;
 use Phake;
-
+use Symfony\Component\Routing\RouteCollection;
```
Comment (alavieille) [inline FrontBundle/Tests/Routing/Database/Transformer/RouteDocumentToValueObjectTransformerTest.php:8]: add an empty line after use

File: FrontBundle/Tests/Routing/Database/Transformer/RouteDocumentToValueObjectTransformerTest.php [hunk 3]
```diff
@@ -21,7 +21,16 @@ class RouteDocumentToValueObjectTransformerTest extends AbstractBaseTestCase
      */
     public function setUp()
     {
-        $this->transformer = new RouteDocumentToValueObjectTransformer();
+        $route = Phake::mock('Symfony\Component\Routing\Route');
```
Comment (alavieille) [inline FrontBundle/Tests/Routing/Database/Transformer/RouteDocumentToValueObjectTransformerTest.php:25]: add a test for the case with NoRenderingMethodForNodeException

File: FrontBundle/Tests/Routing/DatabaseRouteLoaderTest.php [hunk 4]
```diff
@@ -35,7 +35,16 @@ public function setUp()
...
+        $route = Phake::mock('Symfony\Component\Routing\Route');
+        Phake::when($route)->getDefault('_controller')->thenReturn('OpenOrchestra\FrontBundle\Controller\NodeController::showAction');
+
+        $routeCollection = Phake::mock('Symfony\Component\Routing\RouteCollection');
+        Phake::when($routeCollection)->get('open_orchestra_front_node')->thenReturn($route);
+
+        $router = Phake::mock('Symfony\Component\Routing\RouterInterface');
+        Phake::when($router)->getRouteCollection()->thenReturn($routeCollection);
+
+        $this->loader = new DatabaseRouteLoader($this->nodeRepository, $this->siteRepository, $router);
```
Comment (alavieille) [inline FrontBundle/Tests/Routing/DatabaseRouteLoaderTest.php:47]: add a test for the case with NoRenderingMethodForNodeException

**Comment by veolys-nanne** [comment]
update done @itkg-ngilain , @alavieille

**Comment by ngilain** [comment]
:+1:

**Comment by alavieille** [comment]
:+1:

**Pull request merged**